mbed_lstools/lstools_linux_generic.py

```diff
@@ -16,290 +16,89 @@
-            # If TargetID from USBID is not recognized we will try to check URL in mbed.htm
-            mbed_htm_target_id = self.get_mbed_htm_target_id(device[2]) if device[2] else None # device[2] is a 'mount_point'
-            if mbed_htm_target_id is not None:
-                mbed_htm_target_id_prefix = mbed_htm_target_id[0:4]
-                if mbed_htm_target_id_prefix in tids:
-                    # We need to update platform_name and corresponding TargetID (not USBID, but from mbed.htm)
-                    mbed['platform_name'] = self.plat_db.get(mbed_htm_target_id_prefix)
-                    mbed['target_id'] = mbed_htm_target_id
-            mbed['target_id_usb_id'] = tid
-            mbed['target_id_mbed_htm'] = mbed_htm_target_id
-            if mbed['mount_point'] != None or self.list_unmounted:
-                result.append(mbed)
-            else:
-                logger.warning(
-                    "MBED with target id '%s' is connected, but not mounted. "
-                    "Use the '-u' flag to include it in the list.",
-                    mbed['target_id'])
-
-            if None in mbed:
-                self.ERRORLEVEL_FLAG = -1
-
-        return result
-
-    # Private methods
-
-    def get_dev_by_id_cmd(self, subdir):
-        """! Calls command line 'ls' to get devices by their ids
-        @details Uses Linux shell command: 'ls -oA /dev/disk/by-id/'
-        @return tuple(stdout lines, retcode)
-        """
-        cmd = 'ls -oA /dev/' + subdir + '/by-id/'
-        _stdout, _, retval = self.run_cli_process(cmd)
-        return (_stdout.splitlines(), retval)
-
-    def get_dev_by_id_process(self, lines, retval):
-        """! Remove unnecessary lines from command line output
-        """
-        result = []
-        if not retval:
-            for line in lines:
-                line = line.rstrip()
-                if type(line) == type(bytes()):
-                    line = line.decode('utf8')
-                if not line.lower().startswith('total '):    # total 0
-                    result.append(line)
-                    logger.debug("get_dev_by_id_process line %s", line)
-        return result
-
-    def get_dev_by_id(self, subdir):
-        """! Lists disk devices by id
-        @return List of strings from 'ls' command executed in shell
-        """
-        lines, retval = self.get_dev_by_id_cmd(subdir)
-        return self.get_dev_by_id_process(lines, retval)
-
-    def get_mounts(self):
+        dir = join("/dev", device_type, "by-id")
+        if isdir(dir):
+            to_ret = dict(self._hex_ids([join(dir, f) for f in listdir(dir)]))
+            logger.debug("Found %s devices by id %r", device_type, to_ret)
+            return to_ret
+        else:
+            logger.error("Could not get %s devices by id. "
+                         "This could be because your Linux distribution "
+                         "dose not use udev, or does not create /dev/<type>/by-id "
```

Comment:
`dose` -> `dose` plz 😄 

Also, can you substitute the actual `device_type` variable for the `"<type>"` in this line? Just so its more clear for users.